_rules/SC2-4-6-descriptive-labels.md

```diff
@@ -76,35 +89,73 @@ Label that is coded with the `p` element and associated by the aria-labelledby a
 <input aria-labelledby="label_fname" type="text" name="fname"/>
 ```
 
-#### Passed example 3
+#### Passed Example 3
 
 Implicit label that is coded with the `label` element and describes the purpose of the associated element.
 
 ```html
 <label>First name:<input id="fname" type="text" name="fname"/></label>
 ```
 
-#### Passed example 4
+#### Passed Example 4
 
-Label is visible, but not included in accessibility tree
+Label is visible, but not included in accessibility tree.
 
 ```html
 <p id="label_fname" aria-hidden="true">First name:</p>
 <input aria-labelledby="label_fname" type="text" name="fname"/>
 ```
 
-#### Passed example 5
+#### Passed Example 5
 
-Label is included in accessibility tree, but not visible
+Label is included in accessibility tree, but not visible.
 
 ```html
 <p id="label_fname" style="position: absolute; top: -9999px; left: -9999px;">First name:</p>
 <input aria-labelledby="label_fname" type="text" name="fname"/>
 ```
 
+#### Passed Example 6
+
+Programatically determinable headings provide a context that differentiates the purpose of the otherwise identically named form fields.
+
+```html
+<h2>Shipping address</h2>
+<label>Name<input id="name" type="text" name="name"/></label>
+<label>Street<input id="street" type="text" name="street"/></label>
+
+<h2>Billing address</h2>
+<label>Name<input id="name" type="text" name="name"/></label>
+<label>Street<input id="street" type="text" name="street"/></label>
+````
+
+#### Passed Example 7
+
+Accessible name created through "aria-label" describes the purpose of the associated element.
+
+```html
+<input aria-label="First name" id="fname" type="text" name="fname"/>
+```
+
+#### Passed Example 8
+
+Otherwise identically named form fields can be differentiated both visually and in a way that is included in the accessibility tree, though different methods are used for each purpose.
+
+```html
+<h2 aria-hidden="true">Shipping address</h2>
+<p id="shipping" style="position: absolute; top: -9999px; left: -9999px;">(Shipping)</p>
+<label>Name<input id="name" type="text" name="name" aria-describedby="shipping" /></label>(Shipping
+<label>Street<input id="street" type="text" name="street" aria-describedby="shipping" /></label>
+
+<h2 aria-hidden="true">Billing address</h2>
+<p id="shipping" style="position: absolute; top: -9999px; left: -9999px;">(Billing)</p>
+<label>Name<input id="name" type="text" name="name" aria-describedby="billing"/></label>
```

Comment:
And while you're at it, following up on @WilcoFiers points on #546 (unfortunately I couldn't make to today's meeting, so I still don't know what was the decision, but I agree with his point) I think it would be nice to use different ids from the ones already used in the shipping part of the example

Reply:
Each test case runs on a unique page, so maybe it's not a problem if on the rule page we have identical ID?

Reply:
I would prefer if we can reuse the same IDs across test cases, so the test cases are more consistent, and the only thing that changes is what we want to illustrate with that particular test case.
As Audrey says, the test cases should be considered separate pages in themselves.

Reply:
I'm totally for using the same ids in different test cases. 
But in this test case, both `name` and `street` are the ids used in different `input` elements. That was what I was suggesting to avoid because it would fail other rules.

Reply:
Ahh, I see! Fixed it both there and in Passed Example 6